### ISSUE TITLE
Automatically set scan_data_type when searching for string

### ISSUE THREAD

**Issue by bkazemi** [scanmem/scanmem#249]
Since the `"` command is only used when searching for strings, it makes sense to automatically set `scan_data_type` to `string` when this command is entered. It wouldn't break existing scripts because it would make the manually entered `option scan_data_type string` command essentially a noop. Thoughts?

**Comment by 12345ieee** [comment]
Well, the `string__handler()` is triggered by `"` followed by space, it indeed seems safe to assume that.

The `option scan_data_type string` would not become a no-op, as it forbids you from scanning numbers anyway.

I vote yes. @sriemer ?

**Comment by 12345ieee** [comment]
Simple implementation at 12345ieee@33a56aa .

**Comment by bkazemi** [comment]
I meant it would make manually entered `scan_data_type string` from old scripts useless but harmless.

Seems like it was a painless fix. Maybe we could also switch over to an appropriate default number type when a number is entered and `scan_data_type` == `string`. Or maybe that is too reaching.

**Comment by 12345ieee** [comment]
That would be more annoying, imagine being a new user that wants to scan for the string "123" and keeps getting bounced around.
Honestly, I don't really use the CLI interface (I send things with `echo -e` for development), I'm not sure of what would help the workflow.

Now that I'm pretty familiar with the code I can easily implement simple request like this, if you have good UX changes, propose.

**Comment by bkazemi** [comment]
This is being typed on my mobile device. Excuse my brevity.

The first thing that could be changed is recognizing the `"` command without a space. 

Another change to consider is command completion à la gdb. This would of course be a lot of new code, however.

**Comment by 12345ieee** [comment]
Recognizing `"` without a space would be not backwards-compatible (as older scripts would search for '(space)string'), I'm not a fan of that.

As for command completion, I had looked into subcommand completion in libreadline (like option->scan_data_type), but it looked like there was no built-in answer and it'd require full manual handling, that is something I do not want to write.

**Comment by bkazemi** [comment]
Yes, a non-space version would require a major bump. I'll write the command completion code, and see how much new code it takes up.

**Comment by 12345ieee** [comment]
I'm definitely interested in that, `scan_data_type` gets annoying to type.
Let me know when you have something I can review/test/look at.

**Comment by sriemer** [comment]
Yep, sub commands should be completed and yes, `option scan_data_type` is too long. I had a custom patch making all data types top level commands but the former maintainer didn't like it. `type <sd_type>` was the best I could get. Sub command completion would still be required for this.

I'm not a fan of automating the scan data type change and being trapped in there afterwards. It would be better if it would be a temporary change just for the current `" ` command. Then you can search for numbers and strings.

**Comment by 12345ieee** [comment]
Uhm, the issue of exiting the string mode immediately is that if you call `list` (for example) the entries are displayed as integers, that is definitely not what you want (there are also issues with reinterpreting the string length as integer flags, but let's forget about that for a moment).

**Comment by bkazemi** [comment]
I think the best way to handle it would be to only set it automatically when it is an initial scan. I don't think it's common for people to search for numbers on string matches and vice versa. Plus we'd have to mix the match output types which would be a lot of new code.
